core/src/validator.rs

```diff
@@ -512,6 +512,8 @@ impl Validator {
         tpu_enable_udp: bool,
         admin_rpc_service_post_init: Arc<RwLock<Option<AdminRpcRequestMetadataPostInit>>>,
     ) -> Result<Self, String> {
+        let start_time = Instant::now();
```

Comment:
Only thing I don't love is that we don't capture time for stuff in `validator/src/main.rs`. This is mostly parsing args, but one item that comes to mind is RPC bootstrap (ie when we need to download a snapshot). However, given that there is so much variability in downloading snapshots and nodes should not be downloading snaps on reboots, I was fine leaving that out